### PR TITLE
feat(discordx): plugins support

### PR DESCRIPTION
**Pull request by samarmeena**
# Plugin

Develop plugins for discordx. You can publish your plugin on NPM. Maintain a single codebase while using them on different bots.

## Create a Plugin

```ts
import { dirname, importx } from "@discordx/importer";
import { Plugin } from "discordx";

export class HelperPlugin extends Plugin {
  async init(): Promise<void> {
    // This section is similar to the bot login section. However, here we are doing this for a plugin.
    await importx(`${dirname(import.meta.url)}/commands/**/*.{js,ts}`);
  }
}
```

## Use a plugin with discordx client

```ts
import { Client, MetadataStorage } from "discordx";

// Initialize the plugin
const helperPlugin = new HelperPlugin({ metadata: MetadataStorage.instance });

// Provide plugins to client
new Client({ plugins: [helperPlugin] });
```

 # To-Do
 - [ ] Explain Plugin abstract class in document

### CODE REVIEW

**Review by VictoriqueMoe — DISMISSED** [review]

File: docs/docs/faq/Errors/Common.md [hunk 1]
```diff
@@ -65,4 +65,4 @@ Error [DISALLOWED_INTENTS]: Privileged intent provided is not enabled or whiteli
 
 Some [Gateway Intents](https://discord.com/developers/docs/topics/gateway#gateway-intents) require approval if your bot is verified. If your bot is not verified, you can toggle those intents to access them. - _Discord_
 
-![Privileged Gateway Intents](../../../static/img/privileged-gateway-intents.png)
+![Privileged Gateway Intents](../../../static/img/privileged-gateway-intents.png))
```
Comment (xmnlz) [inline docs/docs/faq/Errors/Common.md:68]: why did you add the extra parenthesis?

Comment (samarmeena) [reply docs/docs/faq/Errors/Common.md:68]: merge went wrong, thanks for noticing
